addons/account/models/account_move_line.py

```diff
@@ -1289,14 +1290,41 @@ def invalidate_recordset(self, fnames=None):
     def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
         def to_tuple(t):
             return tuple(map(to_tuple, t)) if isinstance(t, (list, tuple)) else t
+
+        # Since the ORM only supports simple `order by` clauses, add a read amls with a specific amount_residual
+        # so that they appear on the bank rec widget first. It is activated by context key when no order is specified.
+        preferred_aml_ids = self.env.context.get('matching_amount_aml_ids')
+        extra_results = []
+        if not order and preferred_aml_ids:
+
+            extra_results = super().browse(preferred_aml_ids).read(fields=fields)
+            if extra_results:
+
+                # Adapt the domain.
+                domain = expression.AND([domain or [], [('id', 'not in', preferred_aml_ids)]])
+
+                # Adapt offset.
+                nb_extra_results = len(extra_results)
+                extra_results = extra_results[offset:]
+                offset = max(offset - nb_extra_results, 0)
+
+                # Adapt the limit.
+                if limit is not None:
+                    extra_results = extra_results[:limit]
+                    limit -= len(extra_results)
+
+                    # Special case to avoid an additional search_read if there is nothing left to match.
+                    if limit == 0:
+                        return extra_results
+
         # Make an explicit order because we will need to reverse it
         order = (order or self._order) + ', id'
         # Add the domain and order by in order to compute the cumulated balance in _compute_cumulated_balance
         contextualized = self.with_context(
             domain_cumulated_balance=to_tuple(domain or []),
             order_cumulated_balance=order,
         )
-        return super(AccountMoveLine, contextualized).search_read(domain, fields, offset, limit, order)
+        return extra_results + super(AccountMoveLine, contextualized).search_read(domain, fields, offset, limit, order)
```

Comment:
post-process after the super call instead to reorder according to matching_amount_aml_ids? Also, put a more specific context key and put this code on the enterprise module instead